docs/general/run-a-node/upgrade-log.md

```diff
@@ -167,17 +167,213 @@ oasis-node consensus submit_tx \
 
 ### Instructions - Before upgrade
 
-_TBD_
+This upgrade will upgrade **Oasis Core** to **version [22.1]** which **no longer
+allows running Oasis Node** (i.e. the `oasis-node` binary) **as root**
+(effective user ID of 0).
+
+Running network accessible services as the root user is extremely bad for
+system security as a general rule. While it would be "ok" if we could drop
+privileges, `syscall.AllThreadsSyscall` does not work if the binary uses `cgo`
+at all.
+
+Nothing in Oasis Node will ever require elevated privileges.
+Attempting to run the `oasis-node` process as the root user will now terminate
+immediately on startup.
+
+While there may be specific circumstances where it is safe to run network
+services with the effective user ID set to 0, the overwhelming majority of cases
+where this is done is a misconfiguration.
+
+If the previous behavior is required, the binary must be run in unsafe/debug
+mode (via the intentionally undocumented flag), and `debug.allow_root` must also
+be set.
+
+[22.1]: https://github.com/oasisprotocol/oasis-core/releases/tag/v22.1
 
 ### Instructions - Upgrade day
 
-_TBD_
+Following steps should be performed on **2022-04-11** only after the network has
+reached the upgrade epoch and has halted:
+
+* Download the genesis file published in the [Damask Upgrade release].
```

Comment:
If the steps below are order-sensitive, I suggest we use numbering:
```suggestion
1. Download the genesis file published in the [Damask Upgrade release].
```

Reply:
Thanks for the suggestion. Updated.